diff --git a/ErdosProblems/Erdos49/PNT/MellinCalculus.lean b/ErdosProblems/Erdos49/PNT/MellinCalculus.lean
index f2fe09c..f611307 100644
--- a/ErdosProblems/Erdos49/PNT/MellinCalculus.lean
+++ b/ErdosProblems/Erdos49/PNT/MellinCalculus.lean
@@ -897,6 +897,8 @@ lemma MellinOfSmooth1a {ν : ℝ → ℝ} (diffν : ContDiff ℝ 1 ν)
       · apply MeasureTheory.Measure.restrict_mono' SsubT.eventuallyLE le_rfl
       have : volume.restrict (Tx ×ˢ Ty) = (volume.restrict Tx).prod (volume.restrict Ty) := by
         rw [Measure.prod_restrict, MeasureTheory.Measure.volume_eq_prod]
+      change Integrable (fun x : ℝ × ℝ ↦ f x.2 * (x.1 : ℂ) ^ (s - 1))
+        (volume.restrict (Tx ×ˢ Ty))
       conv => rw [this]; lhs; intro; rw [mul_comm]
       apply MeasureTheory.Integrable.mul_prod (f := fun x ↦ (x : ℂ) ^ (s - 1))
         (μ := Measure.restrict volume Tx)
diff --git a/PrimeNumberTheoremAnd/Sobolev.lean b/PrimeNumberTheoremAnd/Sobolev.lean
index 34ab917..e956534 100644
--- a/PrimeNumberTheoremAnd/Sobolev.lean
+++ b/PrimeNumberTheoremAnd/Sobolev.lean
@@ -304,9 +304,7 @@ theorem W21_approximation (f : W21) (g : trunc) :
         have d1 : deriv (fun v => h R v * f v) = fun v => h' R v * f v + h R v * f' v :=
           funext (fun v => (l3 v).deriv)
         rw [d1]
-        convert (l5.add l7).deriv using 1
-        · congr 1
-        · ring
+        convert (l5.add l7).deriv using 1 <;> ring
       simp_rw [this, F]
 
     obtain ⟨c1, mg'⟩ := g'.bounded
